.github/workflows/service_test_hdfs.yml

```diff
@@ -173,3 +173,41 @@ jobs:
           OPENDAL_HDFS_ATOMIC_WRITE_DIR: /tmp/atomic_write_dir/opendal/
           OPENDAL_HDFS_NAME_NODE: default
           OPENDAL_HDFS_ENABLE_APPEND: false
+
+  hdfs-default-gcs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup
+        with:
+          need-nextest: true
+
+      - name: Setup java env
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: "11"
+      - name: Setup hadoop env
+        shell: bash
+        run: |
+          curl -LsSf https://dlcdn.apache.org/hadoop/common/hadoop-3.3.5/hadoop-3.3.5.tar.gz | tar zxf - -C /home/runner
+          curl -LsSf -o /workspaces/opendal/tmp/hadoop-3.3.5/share/hadoop/common/lib/gcs-connector-hadoop2-2.2.19-shaded.jar  https://github.com/GoogleCloudDataproc/hadoop-connectors/releases/download/v2.2.19/gcs-connector-hadoop2-2.2.19-shaded.jar
+
+      - name: Test
+        shell: bash
+        working-directory: core
+        run: |
+          export CLASSPATH=$(${HADOOP_HOME}/bin/hadoop classpath --glob)
+          export LD_LIBRARY_PATH=${{ env.JAVA_HOME }}/lib/server:${{ env.HADOOP_HOME }}/lib/native
+          cp ${{ github.workspace }}/fixtures/hdfs/hdfs-site.xml ${{ env.HADOOP_HOME }}/etc/hadoop/hdfs-site.xml
+          cp ${{ github.workspace }}/fixtures/hdfs/core-site.xml ${{ env.HADOOP_HOME }}/etc/hadoop/core-site.xml
+
+          cargo test behavior --features tests,services-hdfs
+        env:
+          HADOOP_HOME: "/home/runner/hadoop-3.3.5"
+          OPENDAL_TEST: hdfs
+          OPENDAL_HDFS_ROOT: /tmp/opendal/
+          OPENDAL_HDFS_NAME_NODE: gs://bucket_name
```

Comment:
TODO (1/2): Need to substitute this with the secreted bucket name in the form of `gs://${bucket_name}`. Not sure if using `op://services/gcs/bucket` will work?

Reply:
After migrate to https://github.com/apache/opendal/pull/4140, this should be easy.

Reply:
Thanks for the heads up. Moved the changes to a new `.github/services/hdfs/hdfs_default_gcs/action.yml` file